src/middlewared/middlewared/plugins/system/vendor/vendor.py

```diff
@@ -0,0 +1,27 @@
+import os
+
+from middlewared.api import api_method
+from middlewared.api.current import NoArgs, NoneReturn, VendorNameResult
+from middlewared.service import Service
+
+
+SENTINEL_FILE_PATH = '/data/.vendor'
+
+
+class VendorService(Service):
+
+    @api_method(NoArgs, VendorNameResult, private=True)
+    def name(self) -> str | None:
+        try:
+            with open(SENTINEL_FILE_PATH, 'r') as file:
```

Comment:
Get with @aiden3c because this file will be json formatted so we can't just read it outright. We'll need to de-serialize it.

Reply:
Is this a mutable file? Should we cache this method?

Reply:
> Is this a mutable file? Should we cache this method?

I'm going to guess that it is mutable and should not be cached given the `unvendor` method removes the file.